addon/-private/validated-changeset.ts

```diff
@@ -851,14 +857,27 @@ export class BufferedChangeset implements IChangeset {
       }
 
       if (result && isObject(result)) {
-        return normalizeObject(result);
-      }
+        result = normalizeObject(result);
+         let content: Content = this[CONTENT];
+
+        if (isObject(content[baseKey]) && !isEmberDataObject(content[baseKey])) {
+          let data: { [key: string]: any } = {}
+          Object.keys(content[baseKey]).forEach(k => {
+            data[k] = this.getDeep(content, `${baseKey}.${k}`)
+          })
+          return Object.assign(data, result);
+        }
```

Comment:
Do we need this because Ember is asking for `user` and not the leaf key?  I was under the assumption that the key passed was the full path (but perhaps was pre Octane)...For example I see  the`key` arg as `cid` but never `user.name` for example.  Just `user` as the argument to `get`.

Overall this is great!  Can we add a comment on what we are trying to do?  What I gather is you are taking the `CHANGES` and merging with the `CONTENT` at that level in the object, correct?

Reply:
Yeah, I thought it did request the full path as well, but it doesn't. If I render `user.name`, the requested key is just `user`. 

We need this "merge" because the `user ` has changed, but only one property in it, causing the return value to be incomplete when having more than one property. 

The change here fetches the actual content and merge the changes on top of it, having complete data for a nested property.

I will add some comments to the code.

Reply:
![Screen Shot 2020-01-15 at 12 53 52 PM](https://user-images.githubusercontent.com/7374640/72470712-2266b580-3796-11ea-8219-c7037132e120.png)

Yeah looks like it traverses to the parent to find the value :( . Perhaps was a recent update, perhaps not.  

In any case, the fix you have here seems like the correct one.